qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -657,6 +657,12 @@ def _common_extend_steps(self, md_template):
             if not new_cols and not new_samples:
                 return
 
+            is_extendable, error_msg = self.can_be_extended(new_samples,
```

Comment:
I was using the same approach as for the `can_be_updated`. I will implement both in the base method and just raise an error since they've to be overwritten by the subclasses.